activerecord/lib/active_record/readonly_attributes.rb

```diff
@@ -46,14 +46,20 @@ def readonly_attribute?(name) # :nodoc:
     end
 
     module HasReadonlyAttributes # :nodoc:
-      [:write_attribute, :_write_attribute].each do |name|
-        define_method(name) do |attr_name, value|
-          if !new_record? && self.class.readonly_attribute?(attr_name.to_s)
-            raise ReadonlyAttributeError.new(attr_name)
-          end
+      def write_attribute(attr_name, value)
+        if !new_record? && self.class.readonly_attribute?(attr_name.to_s)
+          raise ReadonlyAttributeError.new(attr_name)
+        end
+
+        super(attr_name, value)
```

Comment:
We couldn't use zsuper here before because it doesn't work with `define_method`, but now we could:

```suggestion
        super
```

(not a blocker, of course.)

Reply:
Good call.